pkg/commands/componentarchive/remote/remote_test.go

```diff
@@ -135,4 +136,88 @@ var _ = Describe("Remote", func() {
 		Expect(showOpts.Run(ctx, logr.Discard(), testdataFs)).To(HaveOccurred())
 	})
 
+	It("should copy a component descriptor and its blobs from the source repository to the target repository.", func() {
+		baseFs, err := projectionfs.New(osfs.New(), "../")
+		Expect(err).ToNot(HaveOccurred())
+		testdataFs = layerfs.New(memoryfs.New(), baseFs)
+		ctx := context.Background()
+
+		cf, err := testenv.GetConfigFileBytes()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(vfs.WriteFile(testdataFs, "/auth.json", cf, os.ModePerm))
+
+		baseURLSource := testenv.Addr + "/test-source"
+		baseURLTarget := testenv.Addr + "/test-target"
+
+		pushOpts := &remote.PushOptions{
+			OciOptions: options.Options{
+				AllowPlainHttp:     false,
+				RegistryConfigPath: "/auth.json",
+			},
+		}
+		pushOpts.ComponentArchivePath = "./testdata/01-ca-blob"
+		pushOpts.BaseUrl = baseURLSource
+
+		cmd := remote.NewPushCommand(ctx)
+		Expect(cmd)
+
+		Expect(pushOpts.Run(ctx, logr.Discard(), testdataFs)).To(Succeed())
+
+		repos, err := client.ListRepositories(ctx, baseURLSource)
+		Expect(err).ToNot(HaveOccurred())
+
+		componentName := "example.com/component"
+		componentVersion := "v0.0.0"
+
+		sourceRef := baseURLSource + "/component-descriptors/" + componentName
+		Expect(repos).To(ContainElement(Equal(sourceRef)))
+
+		manifest, err := client.GetManifest(ctx, sourceRef+":"+componentVersion)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(manifest.Layers).To(HaveLen(2))
+		Expect(manifest.Layers[0].MediaType).To(Equal(cdoci.ComponentDescriptorTarMimeType),
```

Comment:
we should ehckc if the localFilesystem blob has been correctly converted to a localOciBlob

Reply:
trying to fetch the cd with 
`		var configBlob bytes.Buffer
		Expect(client.Fetch(ctx, sourceRef+":"+componentVersion, manifest.Layers[0], &configBlob)).To(Succeed())

		cd := &v2.ComponentDescriptor{}
		Expect(codec.Decode(configBlob.Bytes(), cd)).To(Succeed())
`
but fails with "error converting YAML to JSON: yaml: control characters are not allowed"
configBlob.String() shows:
<string>: component-descriptor.yaml\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x000000000\x000000000\x000000000\x0000000001051\x0014101020626\x00014255\x00 0\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00ustar\x0000\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x000000000\x000000000\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00{\"meta\":{\"schemaVersion\":\"v2\"},\"component\":{\"name\":\"example.com/component\",\"version\":\"v0.0.0\",\"repositoryContexts\":[{\"baseUrl\":\"eu.gcr.io/gardener-project/components/dev\",\"type\":\"ociRegistry\"},{\"baseUrl\":\"127.0.0.1:62980/test-source\",\"componentNameMapping\":\"urlPath\",\"type\":\"ociRegistry\"}],\"provider\":\"internal\",\"sources\":[],\"componentReferences\":[],\"resources\":[{\"name\":\"myconfig\",\"version\":\"v0.0.0\",\"type\":\"json\",\"relation\":\"local\",\"access\":{\"digest\":\"sha256:ab894987c426bf8d660826c6fa52a1f351a4c4c094f913862be9c76386bcc32f\",\"type\":\"localOciBlob\"}}]}}\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00...